lib/util/merge-confidence/index.ts

```diff
@@ -25,14 +25,47 @@ export const confidenceLevels: Record<MergeConfidence, number> = {
 export function initConfig(): void {
   apiBaseUrl = getApiBaseUrl();
   token = getApiToken();
+  supportedDatasources =
+    parseSupportedDatasourceString() ?? presetSupportedDatasources;
+
   if (!is.nullOrUndefined(token)) {
     logger.debug(`Merge confidence token found for ${apiBaseUrl}`);
   }
 }
 
+export function parseSupportedDatasourceString(): string[] | undefined {
+  const supportedDatasourceString =
+    process.env.RENOVATE_X_MERGE_CONFIDENCE_SUPPORTED_DATASOURCES;
+
+  if (!is.string(supportedDatasourceString)) {
+    return undefined;
+  }
+
+  let parsedDatasourceList: string[] | undefined;
+  try {
+    parsedDatasourceList = JSON.parse(supportedDatasourceString);
```

Comment:
use the parseJson util

Reply:
you mean this?

https://github.com/renovatebot/renovate/blob/7c87f2f9c40f54242ada25e8ce7c4092c50119ae/lib/util/common.ts#L69-L77

i suppose we can just use `JSON5` directly instead of
```ts
    parsedDatasourceList = parseJson(supportedDatasourceString, ".json5");
```

wdyt?